File: 12nd/pro_1.java
import java.util.*;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Vector<Integer> v= new Vector<Integer>();
        int cnt =-1;
        int current_t =0;
        for(int i=0; i<progresses.length; i++){
            int time=(100-progresses[i])/speeds[i];
            if((100-progresses[i])%speeds[i]!=0) time++;
            if(current_t>=time){
                v.set(cnt,v.get(cnt)+1);
            }else{
                cnt++;
                current_t=time;
                v.add(1);
            }
        }
        int[] answer = new int[v.size()];
        for(int i=0; i<v.size(); i++){
            answer[i]=v.get(i);
        }
        return answer;
    }
}

File: graph/pro_1.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<vector<int>> results) {
    int answer = 0;
    vector<vector<bool>> graph(n+1, vector<bool>(n+1,false));
    
    for(vector<int> v : results){
        graph[v[0]][v[1]]=true;
    }
    
    for(int i=1; i<=n; i++){
        for(int j=1; j<=n; j++){
            for(int k=1; k<=n; k++){
                if(graph[j][i]&&graph[i][k])    graph[j][k]=true;
            }
        }
    }
    
    for(int i=1; i<=n; i++){
        int cnt=0;
        for(int j=0; j<=n; j++){
            if(graph[i][j]||graph[j][i])    cnt++;
        }
        if(cnt==n-1)    answer++;
    }
    return answer;
}